module/ec2/main.tf

```diff
@@ -0,0 +1,48 @@
+data "aws_ami" "ubuntu" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["ubuntu/images/hvm-ssd-gp3/ubuntu-noble-24.04-amd64-server-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  filter {
+    name   = "root-device-type"
+    values = ["ebs"]
+  }
+
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+}
+
+data "aws_availability_zones" "azs" {
+  filter {
+    name   = "opt-in-status"
+    values = ["opt-in-not-required"]
+  }
+}
+
+resource "aws_key_pair" "ssh_key" {
+  key_name   = "my-key"
+  public_key = file("/home/amar.zilpe/Documents/learning_platform/Udemy/ssh_key/ssh_key.pub")
```

Comment:
```
resource "null_resource" "read_public_key" {

  triggers = {
    key_file_changed = file("${var.private_key_path}")
  }

  provisioner "local-exec" {
    command = "ssh-keygen -f ${var.private_key_path} -e > public_key.pem"
  }
}

resource "aws_key_pair" "my_key_pair" {
  key_name = var.key_pair
  public_key = file("${path.module}/public_key.pem")

  # ... other configuration options
}
```
This approach keeps the private key secure and leverages Terraform for managing the public key portion. Remember to adjust the paths and commands based on your specific setup.